src/multio/action/average-rate/AverageRate.cc

```diff
@@ -0,0 +1,89 @@
+/*
+ * (C) Copyright 1996- ECMWF.
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ * In applying this licence, ECMWF does not waive the privileges and immunities
+ * granted to it by virtue of its status as an intergovernmental organisation nor
+ * does it submit to any jurisdiction.
+ */
+
+#include "AverageRate.h"
+
+#include "eckit/config/Configuration.h"
+
+#include "multio/LibMultio.h"
+#include "multio/message/Glossary.h"
+
+namespace multio::action::average_rate {
+
+namespace {
+Param2ParamMap getMappings() {
+    eckit::LocalConfiguration mappingConf{eckit::YAMLConfiguration{eckit::PathName{
+        multio::LibMultio::instance().libraryHome() + "/share/multio/config/average_rate_param_mappings.yml"
+    }}};
+
+    Param2ParamMap paramMappings;
+    for (auto& mappings : mappingConf.getSubConfigurations()) {
+        auto paramIn = mappings.getInt64("paramIn");
+        auto paramOut = mappings.getInt64("paramOut");
+        paramMappings[paramIn] = paramOut;
+    }
+    return paramMappings;
+}
+}
+
+AverageRate::AverageRate(const ComponentConfiguration& compConf) :
+    ChainedAction(compConf), paramMappings_{getMappings()} {}
+
+void AverageRate::executeImpl(message::Message msg) {
+    // Skip non-field messages
+    if (msg.tag() != message::Message::Tag::Field) {
+        executeNext(std::move(msg));
+        return;
+    }
+
+    // Skip non-statistics messages
+    if (msg.metadata().getOpt<std::int64_t>(multio::message::Mtg2::mars::timespan).value_or(0) <= 0) {
+        executeNext(std::move(msg));
+        return;
+    }
+
+    // Try to find and apply the mapping for this param, skip otherwise
+    auto param = msg.metadata().get<std::int64_t>(glossary().param);
+    if (auto search = paramMappings_.find(param); search != paramMappings_.end()) {
+        msg.acquire();
+        msg.modifyMetadata().set(glossary().param, search->second);
+    } else {
+        executeNext(std::move(msg));
```

Comment:
Do we want a strict mode where we throw an exception in case the mapping is not found?